### ISSUE TITLE
Composer dependencies

### ISSUE THREAD

**Issue by mlocati** [matt9mg/c5-profiler#2]
This repo currently has 2 `composer.json` files, which is useless IMHO.

I'd create a single `composer.json` file (in the root of the repository, after #1 has been implemented), with the core and package dependencies, that is with something like this:

```json
"require": {
    "concrete5/core": ">=8.5.1",
    "maximebf/debugbar": "^1.16"
}
```
where `"concrete5/core": ">=8.5.1"` reflects the value of the `$appVersionRequired` property of the package controller.

I'd also remove the `composer.lock` file and the `vendor` directory from the repository. That way, other concrete5 packages that require the same composer dependencies (eg `maximebf/debugbar`) still keep working.

This solution is great for composer based installations of concrete5.

But what about non-composer installations? You need to create a package with your composer dependencies, which is not that easy. For example, what if `maximebf/debugbar` requires a composer package that's already included in concrete5? Simply running `composer install` in the package folder would create a vendor directory that includes packages already provided by the core.
To solve this issue, you can use the `composerpkg` command provided by https://github.com/concrete5/cli 
To install the composer dependencies you simply run `composerpkg install` instead of `composer install`.

In your package controller you then need to detect if the package has been installed via composer or not.
- In the first case, autoloading the package dependencies is already working (the core autoloader will load your dependencies too).
- In the second case, you have to configure the autoload for your packages

In other words, in the `on_start` method you need some code like this:

```php
public function on_start()
{
    $this->setupAutoloader();
}

/**
 * Includes the composer autoloader (if this package has not been installed via composer).
 */
private function setupAutoloader()
{
    $path = $this->getPackagePath() . '/vendor/autoload.php';
    if (file_exists($path)) {
        require_once $path;
    }
}
```

PS: for a working example, you can take a look at https://github.com/concrete5-community/acme

**Comment by matt9mg** [comment]
That is a lovely idea :)
